### ISSUE TITLE
Multi-consumer e2e blocker

### ISSUE THREAD

**Issue by smarshall-spitzbart** [cosmos/interchain-security#493]
# Problem

I'm running into odd IBC related issues with e2e tests that setup the CCV channel for multiple in-mem consumers.

# Closing criteria

`TestRelayAndApplySlashPacket` passes with multiple consumers and multiple established CCV channels.

# Problem details

In developing #475, I refactored `CCVTestSuite.SetupTest()` and `CCVTestSuite.SetupCCVChannel()` to be compatible with a list of consumers. There should be no logical changes to individual consumer setup, just for-loops which expand the same setup to multiple consumers. 

The tests pass without error with these new setup refactors (and instantiating 5 consumer chains, apps, and paths). However, when the list contains more than one consumer, and there is more than one call to [SetupCCVChannel](https://github.com/cosmos/interchain-security/blob/e2e-multiple-consumers/tests/e2e/slashing.go#L43) for different consumer paths, the first relayed packet in `TestRelayAndApplySlashPacket` causes this error
```bash
failed to execute message; message index: 0: receive packet verification failed: packet source channel doesn't match the counterparty's channel (channel-1 ≠ channel-0): invalid packet
            	Test:       	TestCCVTestSuite/TestRelayAndApplySlashPacket
```
You can replicate this error by checking out `e2e-multiple-consumers` (which passes as committed) and adding this single line to `TestRelayAndApplySlashPacket`[here](https://github.com/cosmos/interchain-security/blob/e2e-multiple-consumers/tests/e2e/slashing.go#L43)

Line to add:
```go
s.SetupCCVChannel(s.consumerBundles["testchain3"].Path)
```
^ this sets up the second consumer path (chain ids start at "testchain2")

It seems there is some state that is being broken for the first consumer's `ibctesting.Path`, that is only caused by setting up the CCV channel for another consumer. This error could also be showing up if the generated packets from the provider CCV module are incorrectly referencing channel IDs for specific consumers. My hunch is that the problem lies in e2e setup and not the CCV module, but that hunch is unconfirmed. Any help is appreciated!

# TODOs
- [x] Labels have been added for issue
- [x] Issue has been added to the ICS project

**Comment by smarshall-spitzbart** [comment]
Note this could have something to do with just arbitrary string values that're being set in `ibctesting`, that do not account for multi-link IBC topologies

**Comment by Abhishek1857** [comment]
@smarshall-spitzbart how did you added proposal for consumer chain addition. Can you please tell .

**Comment by smarshall-spitzbart** [comment]
Update: the bug is being caused by the ibc go testing package (from the beta version of ibc go we reference). 

`ibctesting.TestChain.GetSentPacket(sequence uint64)` is the api that is exposed for a test chain to obtain a sent packet (that is stored via events). This method indexes sent packets via their seq number, but no form of channelID. Therefore, if we try to setup multiple IBC connections from one chain, stored sent packets will be overwritten and not all accessible.

**Comment by smarshall-spitzbart** [comment]
@danwt I have a super hacky fix for this issue on #475, but I'm sure we can consolidate the functionality I need into `/testutil/simibc`! Let's collab next week

**Comment by danwt** [comment]
FYI GetSentPacket exists in our fork of ibc-go but not ibc-go itself. https://github.com/informalsystems/ibc-go/commit/c48f8e2dadb897d91017df43bd0f03719889637b

simibc doesn't use this! 

I agree, let's collab 👍
